crates/hir-ty/src/infer/coerce.rs

```diff
@@ -626,6 +620,7 @@ impl InferenceTable<'_> {
             }
             _ => None,
         };
+        let must_be_unsized = must_be_coerce_unsized(from_ty, to_ty);
```

Comment:
Let's move this call into the match arm, since its only used there

Reply:
This is not possible as it's recursive.